src/extension.ts

```diff
@@ -1,21 +1,76 @@
-import * as vscode from 'vscode';
+import { exec } from 'child_process';
+import * as fs from 'fs/promises';
 import * as os from 'os';
 import * as path from 'path';
-import { exec } from 'child_process';
 import { promisify } from 'util';
+import * as vscode from 'vscode';
 import {
+    Executable,
     LanguageClient,
     LanguageClientOptions,
     ServerOptions,
-    TransportKind,
-    Executable
+    TransportKind
 } from 'vscode-languageclient/node';
+import { which } from './which';
 
 let client: LanguageClient;
 
 const execAsync = promisify(exec);
 const MIN_ZIZMOR_VERSION = '1.11.0';
 
+async function getZizmor(config: vscode.WorkspaceConfiguration): Promise<string | undefined> {
+    const rawConfiguredPath = config.get<string>('executablePath');
+
+    if (rawConfiguredPath) {
+        const configuredPath = await which(expandTilde(rawConfiguredPath));
+
+        if (configuredPath) {
+            console.log(`Using configured zizmor: ${configuredPath}`);
+            return configuredPath;
+        } else {
+            console.warn(`Configured zizmor not found: ${rawConfiguredPath}`);
+        }
+    }
+
+    const pathPath = await which('zizmor');
+    if (pathPath) {
+        console.log(`Using zizmor from PATH: ${pathPath}`);
+        return pathPath;
+    }
+
+    // It's particularly important to check common locations on macOS because of https://github.com/microsoft/vscode/issues/30847#issuecomment-420399383.
```

Comment:
I have no idea what I'm doing right (or wrong), but I've haven't hit these kinds of problems in VS Code with macOS -- zizmor is installed globally via `brew`, and my extension consistently picks it up via PATH discovery. However, others seem to hit this a lot, so this set of common paths seems reasonable to me.

(I launch it graphically as well, so it's not an interactive shell inheritance thing.)